torch/_prims/__init__.py

```diff
@@ -2528,6 +2529,64 @@ def _svd_aten(
 # Randomness Prims
 #
 
+# TODO: add generator support
+# NOTE: there is currently no way of acquiring the "default" torch generator
+def _normal_meta(
+    shape: ShapeType,
+    *,
+    mean: Union[float, complex],
+    std: float,
+    dtype: torch.dtype,
+    device: torch.device,
+    requires_grad: bool,
+) -> TensorLikeType:
+    utils.check(
+        std >= 0.0,
```

Comment:
if you plan to add refs.normal, this check should go to ref?

Reply:
Possibly. I was thinking we should be a little clearer about what checks go in prims or not. Prims don't verify that their inputs are the expected Python type, but they do validate input metadata, which could include the values of scalar arguments.